staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -376,17 +376,20 @@ func fieldInfoFromField(structType reflect.Type, field int) *fieldInfo {
 	typeField := structType.Field(field)
 	jsonTag := typeField.Tag.Get("json")
 	if len(jsonTag) == 0 {
-		// Make the first character lowercase.
-		if typeField.Name == "" {
+		if info.name == "" && !typeField.Anonymous {
```

Comment:
I'm feeling confused.  When will `info.name` (zero value initialized above) have a `.name` other than `""`?

Reply:
So it is spurious/unnecessary?  Also, why not us the more common `len` check in our codebase?

Reply:
rebased, dropped this unnecessary check and switched the other one to a `len` check